pkg/services/sso/redhatsso_service.go

```diff
@@ -0,0 +1,136 @@
+	// TODO: implement
+	return "", nil
+}
+
+func (r *redhatssoService) RegisterOSDClusterClientInSSO(accessToken string, clusterId string, clusterOathCallbackURI string) (string, *errors.ServiceError) {
+	// TODO: implement
+	return "", nil
+}
+
+func (r *redhatssoService) DeRegisterClientInSSO(accessToken string, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil {
+		return errors.NewWithCause(errors.ErrorFailedToDeleteSSOClient, err, "failed to delete the sso client")
+	}
+	glog.V(5).Infof("Kafka Client %s deleted successfully", clientId)
+	return nil
+}
+
+func (r *redhatssoService) GetConfig() *keycloak.KeycloakConfig {
+	return nil
+}
+
+func (r *redhatssoService) GetRealmConfig() *keycloak.KeycloakRealmConfig {
+	return nil
+}
+
+func (r *redhatssoService) IsKafkaClientExist(accessToken string, clientId string) *errors.ServiceError {
+	return nil
+}
+
+func (r *redhatssoService) CreateServiceAccount(accessToken string, serviceAccountRequest *api.ServiceAccountRequest, ctx context.Context) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.CreateServiceAccount(accessToken, serviceAccountRequest.Name, serviceAccountRequest.Description)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorFailedToCreateServiceAccount, err, "failed to create service account")
+	}
+
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) DeleteServiceAccount(accessToken string, ctx context.Context, clientId string) *errors.ServiceError {
+	err := r.client.DeleteServiceAccount(accessToken, clientId)
+	if err != nil { //5xx
+		return errors.NewWithCause(errors.ErrorFailedToDeleteServiceAccount, err, "failed to delete service account")
+	}
+	glog.V(5).Infof("deleted service account clientId = %s ", clientId)
+	return nil
+}
+
+func (r *redhatssoService) ResetServiceAccountCredentials(accessToken string, ctx context.Context, clientId string) (*api.ServiceAccount, *errors.ServiceError) {
+	serviceAccount, err := r.client.RegenerateClientSecret(accessToken, clientId)
+	if err != nil { //5xx
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to reset service account credentials")
+	}
+	return convertServiceAccountDataToAPIServiceAccount(&serviceAccount), nil
+}
+
+func (r *redhatssoService) ListServiceAcc(accessToken string, ctx context.Context, first int, max int) ([]api.ServiceAccount, *errors.ServiceError) {
+	serviceAccounts, err := r.client.GetServiceAccounts(accessToken, first, max)
+	if err != nil {
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to collect service accounts")
+	}
+
+	var saList []api.ServiceAccount
+	for _, sa := range serviceAccounts {
+		saList = append(saList, *convertServiceAccountDataToAPIServiceAccount(&sa))
+	}
+	return saList, nil
+}
```

Comment:
@akoserwal old API was filtering the service accounts by `rh-org-id`. Is it ok with the new API to just get the list without any filter?
Another check was done in the previous API that the `clientId` was starting with `srvc-acct-`: do we have to do that check with the new API?

Reply:
Leave them //todo for now.

Reply:
Separate jira to work on it